service-providers/network-requester/src/cli/mod.rs

```diff
@@ -111,6 +117,36 @@ fn try_upgrade_v1_1_13_config(id: &str) -> std::io::Result<()> {
     updated.save_to_file(None)
 }
 
+// this only checks compatibility between config the binary. It does not take into consideration
+// network version. It might do so in the future.
+fn version_check(cfg: &Config) -> bool {
```

Comment:
I think those in-binary version checks are rather useless at this point, but I guess that's a topic for another discussion